libs/vkd3d/command.c

```diff
@@ -3442,21 +3442,19 @@ static void d3d12_command_list_end_current_render_pass(struct d3d12_command_list
                 list->so_counter_buffers, list->so_counter_buffer_offsets));
     }
 
-    if (list->current_render_pass)
+    if (list->rendering_info.state_flags & VKD3D_RENDERING_ACTIVE)
     {
-        VkSubpassEndInfoKHR subpass_end_info;
-        subpass_end_info.sType = VK_STRUCTURE_TYPE_SUBPASS_END_INFO_KHR;
-        subpass_end_info.pNext = NULL;
+        list->rendering_info.state_flags &= ~VKD3D_RENDERING_ACTIVE;
 
-        VK_CALL(vkCmdEndRenderPass2KHR(list->vk_command_buffer, &subpass_end_info));
+        VK_CALL(vkCmdEndRenderingKHR(list->vk_command_buffer));
     }
 
     /* Don't emit barriers for temporary suspension of the render pass */
-    if (!suspend && (list->current_render_pass || list->render_pass_suspended))
+    if (!suspend && (list->rendering_info.state_flags & (VKD3D_RENDERING_ACTIVE | VKD3D_RENDERING_SUSPENDED)))
```

Comment:
I'm a little bit confused about this check. VKD3D_RENDERING_ACTIVE is always cleared above, and we still check for it here. In the old path, list->current_render_pass was not set to NULL_HANDLE.

Reply:
Yeah the reset should happen later.

Reply:
fixed.